src/routes/withdraw/withdraw-form.tsx

```diff
@@ -69,6 +69,11 @@ export const WithdrawForm = ({
     }
   }, [status, history]);
 
+  React.useMemo(async () => {
```

Comment:
You are using useMemo like a useEffect

Reply:
Also, I'm not sure we need to call ethers.utils.isAddress in two places. Can we have a single useMemo for `addressValid`? I'm not sure we need an additional useState for storing its validity.

Then we can return early in on submit if `!addressValid` and additionally prevent submit at all by disabling the button?

Reply:
changed